charts/prometheus/templates/prometheus-alerts-configmap.yaml

```diff
@@ -112,107 +121,769 @@ data:
+              >
+            (count(kube_node_status_allocatable_memory_bytes)-1)
+              /
+            count(kube_node_status_allocatable_memory_bytes)
           for: 5m
           labels:
-            tier: platform
-            component: registry
+            severity: warning
+        - alert: KubeCPUOvercommit
           annotations:
-            summary: "Docker Registry High Disk Usage"
-            description: "Docker Registry has less than 10% disk space available."
-
-        - alert: ElasticsearchDiskUsage
-          expr: (sum(kubelet_volume_stats_available_bytes{persistentvolumeclaim=~"data-{{ .Release.Name }}-elasticsearch-data-.*"}) / sum(kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~"data-{{ .Release.Name }}-elasticsearch-data-.*"}) * 100) < 10
+            message: Cluster has overcommitted CPU resource requests for Namespaces.
+            runbook_url:
+          expr: |
+            sum(kube_resourcequota{job="kube-state", type="hard", resource="cpu"})
+              /
+            sum(kube_node_status_allocatable_cpu_cores)
+              > 1.5
           for: 5m
           labels:
-            tier: platform
-            component: elasticsearch
+            severity: warning
+        - alert: KubeMemOvercommit
           annotations:
-            summary: "Elasticsearch High Disk Usage"
-            description: "Elasticsearch has less than 10% disk space available."
-
-        - alert: IngessCertificateExpiration
-          expr: avg(nginx_ingress_controller_ssl_expire_time_seconds) by (host) - time() < 604800
-          for: 10s
+            message: Cluster has overcommitted memory resource requests for Namespaces.
+            runbook_url:
+          expr: |
+            sum(kube_resourcequota{job="kube-state", type="hard", resource="memory"})
+              /
+            sum(kube_node_status_allocatable_memory_bytes{job="node-exporter"})
+              > 1.5
+          for: 5m
           labels:
-            tier: platform
-            component: ingress
+            severity: warning
+        - alert: KubeQuotaExceeded
           annotations:
-            summary: "TLS Certificate Expiring Soon"
-            {{/* We want '{{ $labels.host }}' to be evaluted by prometheus, not helm, so we have to escape it once */ -}}
-            description: {{ printf "%q" "The TLS Certificate for {{ $labels.host }} is expiring in less than a week." }}
+            message: Namespace {{`{{ $labels.namespace }}`}} is using {{`{{ $value | humanizePercentage
+              }}`}} of its {{`{{ $labels.resource }}`}} quota.
+            runbook_url: 
+          expr: |
+            kube_resourcequota{job="kube-state", type="used"}
+              / ignoring(instance, job, type)
+            (kube_resourcequota{job="kube-state", type="hard"} > 0)
+              > 0.90
+          for: 15m
+          labels:
+            severity: warning
+        - alert: CPUThrottlingHigh
```

Comment:
We might want to constrain this one to the release namespace

Reply:
Is that so we exclude certain things? If so what, I don’t fully follow what constrain to the namespace means.

Reply:
I updated this rule to only look at non-customer namespaces.